### PR TITLE
Fix SDL loading policy db from backup

### LINKED ISSUE #2427: SDL changes data in Policy database in case IgnitionOff/On cycle
### Bug Report
SDL changes data in Policy database in case IgnitionOff/On cycle

##### Reproduction Steps
1. Make sure there is no `Request type` defined for `device` in `app_policies` of PreloadedPT
2. Start SDL
3. Register app
4. Check Policy database => there are `device` and `pre_DataConsent` items in `application` table
5. Perform IgnitionOff/On cycle
6. Check Policy database one more time

##### Expected Behavior
SDL does not change data in Policy database during IgnitionOff/On cycle

##### Observed Behavior
Policy database has updated:
1. values for `device` in `application` table has changed:
- `default_hmi`:  NONE => BACKGROUND
- `priority_value`: NONE => EMERGENCY
2. `pre_DataConsent` item has removed at all (only for EXTERNAL_PROPRIETARY)

There is the following error in SDL log:
```
DEBUG [2018-07-28 11-11:41,334][0x7fedf3fff700][Policy] /home/db/ramdrv/sdl_core/src/components/policy/policy_external/src/policy_manager_impl.cc:1297 PolicyManagerImpl::IsPTValid: Errors: policy_table.policy_table.app_policies.device: object is not initialized
policy_table.policy_table.app_policies.device.groups: object is not initialized
policy_table.policy_table.app_policies.device.priority: value is not initialized
policy_table.policy_table.app_policies.device.default_hmi: value is not initialized
policy_table.policy_table.app_policies.device.keep_context: value is not initialized
policy_table.policy_table.app_policies.device.steal_focus: value is not initialized
ERROR [2018-07-28 11-11:41,334][0x7fedf3fff700][Policy] /home/db/ramdrv/sdl_core/src/components/policy/policy_external/src/policy_manager_impl.cc:499 PolicyManagerImpl::RequestPTUpdate: Invalid Policy table snapshot - PTUpdate failed
```

##### OS & Version Information
* OS/Version: Ubuntu 16.04
* SDL Core Version: https://github.com/smartdevicelink/sdl_core/tree/develop
* Testing Against: ATF

##### Test Case, Sample Code, and / or Example App
* [Smoke/Resumption/007_Resumption_FULL_IGNITION_OFF](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Smoke/Resumption/007_Resumption_FULL_IGNITION_OFF.lua)
* [Policies/user_consent_of_Policies/221_ATF_Factory_reset](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Policies/user_consent_of_Policies/221_ATF_Factory_reset.lua)

##### Notes:
- Issue is introduced by https://github.com/smartdevicelink/sdl_core/pull/2149 and not reproduced before
- Issue is reproduced for all SDL flows: PROPRIETARY, HTTP, EXTERNAL_PROPRIETARY

### PR DESCRIPTION
**Pull request by AByzhynar**
Fixes #2427 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Testing Plan
ATF scripts: 
* [Smoke/Resumption/007_Resumption_FULL_IGNITION_OFF](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Smoke/Resumption/007_Resumption_FULL_IGNITION_OFF.lua)
* [Policies/user_consent_of_Policies/221_ATF_Factory_reset](https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/develop/test_scripts/Policies/user_consent_of_Policies/221_ATF_Factory_reset.lua)

### Summary
Added checks for absence of request types and request subtypes  and handling such cases

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/policy/policy_external/src/sql_pt_ext_representation.cc [hunk 1]
```diff
@@ -879,6 +879,9 @@ bool SQLPTExtRepresentation::GatherApplicationPoliciesSection(
     if (!GatherRequestType(app_id, &*params.RequestType)) {
       return false;
     }
+    if (!GatherRequestSubType(app_id, &*params.RequestSubType)) {
```
Comment (EKuliiev) [inline src/components/policy/policy_external/src/sql_pt_ext_representation.cc]: @a-byzhyn
Don't you think the changes
if (!GatherRequestSubType(app_id, &*params.RequestSubType)) {
return false;
}

should be added also to GatherApplicationPoliciesSection funtion in regular_policy?

Comment (EKuliiev) [reply src/components/policy/policy_external/src/sql_pt_ext_representation.cc]: @AByzhynar 
return false means something wrong with the db.
RequestSubType is an optional parameters in the RPC.
Does the function GatherRequestSubType take that factor into consideration?

Comment (AByzhynar) [reply src/components/policy/policy_external/src/sql_pt_ext_representation.cc]: @EKuliiev `params` is output parameter.

**Comment by AByzhynar** [comment]
Rebuild required

**Comment by jacobkeeler** [comment]
Fairly certain that this was already fixed separately with https://github.com/smartdevicelink/sdl_core/pull/2609

**Comment by JackLivio** [comment]
@AByzhynar ^

**Comment by AByzhynar** [comment]
@jacobkeeler @JackLivio Can I close this PR as not actual anymore? And of course I will remove this fix from related PR #2494

**Comment by dboltovskyi** [comment]
Issue is fixed by https://github.com/smartdevicelink/sdl_core/pull/2609

**Pull request closed without merging**